nilearn/input_data/nifti_spheres_masker.py

```diff
@@ -0,0 +1,237 @@
+        the number of functions that will be cached. Zero means no caching.
+
+    verbose: integer, optional
+        Indicate the level of verbosity. By default, nothing is printed.
+
+    See also
+    ========
+    nilearn.input_data.NiftiMasker
+    """
+    # memory and memory_level are used by CacheMixin.
+
+    def __init__(self, seeds, radius=None, smoothing_fwhm=None, standardize=False,
+                 detrend=False, low_pass=None, high_pass=None, t_r=None,
+                 memory=Memory(cachedir=None, verbose=0), memory_level=1,
+                 verbose=0):
+        self.seeds = seeds
+        self.radius = radius
+
+        # Parameters for _smooth_array
+        self.smoothing_fwhm = smoothing_fwhm
+
+        # Parameters for clean()
+        self.standardize = standardize
+        self.detrend = detrend
+        self.low_pass = low_pass
+        self.high_pass = high_pass
+        self.t_r = t_r
+
+        # Parameters for joblib
+        self.memory = memory
+        self.memory_level = memory_level
+
+        self.verbose = verbose
+
+    def fit(self, X=None, y=None):
+        """Prepare signal extraction from regions.
+
+        All parameters are unused, they are for scikit-learn compatibility.
+        """
+        if isinstance(self.seeds, basestring) or is_img(self.seeds):
+            # Take the coordinates of seeds in native space
+            seeds_img = check_niimg(self.seeds, ensure_3d=True)
+            seeds = list(np.where(seeds_img.get_data()))
+            seeds.append(np.ones(len(seeds[0])))
+            seeds = np.asarray(seeds)
+            self.seeds_ = np.dot(seeds_img.get_affine(), seeds)[:3].T
+        else:
+            # This is not elegant but this is the easiest way to test it.
+            try:
+                for seed in self.seeds:
+                    assert(len(seed) == 3)
+            except Exception as e:
+                if self.verbose > 0:
+                    print('Seeds not valid, error' + str(e))
+                raise ValueError('Seeds must be a list of triplets of '
+                                 'coordinates in native space.')
+            self.seeds_ = self.seeds
+        return self
+
+    def fit_transform(self, imgs, confounds=None):
+        return self.fit().transform(imgs, confounds=confounds)
+
+    def _check_fitted(self):
+        if not hasattr(self, "seeds_"):
+            raise ValueError('It seems that %s has not been fitted. '
+                             'You must call fit() before calling transform().'
+                             % self.__class__.__name__)
+
+    def transform(self, imgs, confounds=None):
+        """Extract signals from images.
+
+        Parameters
+        ==========
+        imgs: Niimg-like object
+            See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg.
+            Images to process. It must boil down to a 4D image with scans
```

Comment:
nibabel.get_data() automatically makes sure that time dimension comes last. Not sure the related part in the imgs-explanation is necessary.